tests/http_client_mock.py

```diff
@@ -185,17 +185,27 @@ def assert_post_data(self, expected, is_json=False):
 
 
 class HTTPClientMock(object):
-    def __init__(self, mocker, is_streaming=False):
-        self.mock_client = mocker.Mock(
-            wraps=stripe.http_client.new_default_http_client()
-        )
+    def __init__(self, mocker, is_streaming=False, is_async=False):
```

Comment:
I wonder if we should keep adding more HTTPClientMock instances of store the information about the call in one instance of mocker (e.g. `(verb, path, ..., streaming, async)`).

Reply:
That way the same test can mix multiple kinds of call as well.

Reply:
I can file this as a potential follow-up.